testing/sample_framework/src/android/android_app_framework.cc

```diff
@@ -478,6 +479,40 @@ std::string ReadTextInput(const char* title, const char* message,
   return g_text_entry_field_data->ReadText(title, message, placeholder);
 }
 
+void SetEnvironmentVariableFromExtra(const char* extra_name, JNIEnv* env,
+                                     jobject intent,
+                                     jmethodID get_string_extra) {
+  jstring extra_value_jstring = (jstring)env->CallObjectMethod(
```

Comment:
I forget the jni rules for object ownership lifetimes. I see that you called ReleaseStringUTFChars below, but do you also have to call DeleteLocalRef on this as well?

Reply:
Done